### PR TITLE
something terrible, just to test

### PR DESCRIPTION
**Pull request by aliliolek**
[DEMO LINK](https://aliliolek.github.io/react_autocomplete/)

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -1,55 +1,163 @@
...
 import './App.scss';
+import classNames from 'classnames';
 import { peopleFromServer } from './data/people';
+import { DropdownList } from './components/DropdownList';
+import { Person } from './types/Person';
 
 export const App: React.FC = () => {
-  const { name, born, died } = peopleFromServer[0];
+  const [selectedPerson, setSelectedPerson]
+    = useState<Person | null>(peopleFromServer[0]);
```
Comment (maxim2310) [inline src/App.tsx:16]: you must not select person by default 
image

File: src/App.tsx [hunk 2]
```diff
@@ -1,55 +1,163 @@
...
+  const [query, setQuery] = useState('');
+  const [visibleQuery, setVisibleQuery] = useState('');
+
+  const toggleDropdown = () => {
+    setClickCount(clickCount + 1);
+    setIsActive(clickCount % 2 === 0);
+  };
+
+  const applyQuery = useCallback(
+    debounce(setQuery, 1000),
```
Comment (maxim2310) [inline src/App.tsx:28]: make this delay is smaller, about 0.8s or 0.5s

File: src/App.tsx [hunk 3]
```diff
@@ -1,55 +1,163 @@
...
+        </h1>
+        {selectedPerson && (
+          <a
+            href="./"
+            onClick={(event) => {
+              event.preventDefault();
+              onClose();
+              setVisibleQuery('');
+              setQuery('');
+            }}
```
Comment (maxim2310) [inline src/App.tsx:74]: to much actions, move to separate func

File: src/App.tsx [hunk 4]
```diff
@@ -1,55 +1,163 @@
...
+            className="close-button close-link icon-link"
+          >
+            <IconClose
+              className={classNames(
+                'icon',
+                'is-small',
+                'icon--close',
+              )}
```
Comment (maxim2310) [inline src/App.tsx:82]: why do you need classnames here?

File: src/App.tsx [hunk 5]
```diff
@@ -1,55 +1,163 @@
...
-
-            <div className="dropdown-item">
-              <p className="has-text-danger">Elisabeth Hercke</p>
-            </div>
-          </div>
+        <div
+          className="
+            dropdown-menu
+            dropdown-menu--height
+            "
```
Comment (maxim2310) [inline src/App.tsx:154]:
```suggestion
          className="dropdown-menu dropdown-menu--height"
```

**Review by maxim2310 — CHANGES_REQUESTED** [review]
Great job overall! Just couple tiny things needs to be fixed before approval)

File: src/App.tsx [hunk 6]
```diff
@@ -1,55 +1,154 @@
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import debounce from 'lodash.debounce';
```
Comment (mykhalenych) [inline src/App.tsx:2]: Screenshot 2023-09-05 at 17 31 28
you are using lodash but didn't install module in package.json
also try to write your own debounce function

File: src/App.tsx [hunk 7]
```diff
@@ -1,55 +1,154 @@
...
+  const [selectedPerson, setSelectedPerson]
+    = useState<Person | null>(null);
+  const [isActive, setIsActive] = useState(false);
+  const [clickCount, setClickCount] = useState(0);
+  const [query, setQuery] = useState('');
+  const [visibleQuery, setVisibleQuery] = useState('');
+
+  const toggleDropdown = () => {
+    setClickCount(clickCount + 1);
+    setIsActive(clickCount % 2 === 0);
```
Comment (mykhalenych) [inline src/App.tsx:24]: why do you need set addition state after toggle?
`const isActive = clickCount % 2 === 0`

**Review by mykhalenych — CHANGES_REQUESTED** [review]
Nice work, but fix small suggestions

**Pull request closed without merging**